File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-core/crypto/asymmetric/AsymKeyPair.js
/**
 * @constructor
 * @param publicKey {String}
 * @param privateKey {String}
 * @return {AsymKeyPair}
 */
function AsymKeyPair(publicKey = null, privateKey = null) {
    /**
     * @type String
     */
    this._publicKey = publicKey
    /**
     * @type String
     */
    this._privateKey = privateKey

}


/**
 * @return String
 */
AsymKeyPair.prototype.getPublicKey = function () {
    return this._publicKey
}
/**
 * @param publicKey String
 * @return void
 */
AsymKeyPair.prototype.setPublicKey = function (publicKey) {
    this._publicKey = publicKey
}
/**
 * @return String
 */
AsymKeyPair.prototype.getPrivateKey = function () {
    return this._privateKey
}
/**
 * @param privateKey String
 * @return void
 */
AsymKeyPair.prototype.setPrivateKey = function (privateKey) {
    this._privateKey = privateKey
}

/**
 *
 * @param pair {AsymKeyPair}
 * @return {AsymKeyPair}
 */
AsymKeyPair.copy=function(pair){
    return new AsymKeyPair(pair.getPublicKey(),pair.getPrivateKey())
}

/**
 *
 * @return {AsymKeyPair}
 */
AsymKeyPair.prototype.copy=function(){
    return AsymKeyPair.copy(this)
}

export default AsymKeyPair

File: i2f-jdk/i2f-crypto-std/src/main/java/i2f/crypto/std/encrypt/asymmetric/key/AsymKeyPair.java
package i2f.crypto.std.encrypt.asymmetric.key;

import i2f.codec.str.obfuscate.Base64Obfuscator;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.*;
import java.security.NoSuchAlgorithmException;
import java.security.spec.InvalidKeySpecException;

/**
 * @author Ice2Faith
 * @date 2023/9/5 22:17
 * @desc
 */
@Data
@NoArgsConstructor
public class AsymKeyPair {
    private String publicKey;
    private String privateKey;

    public AsymKeyPair(String publicKey, String privateKey) {
        this.publicKey = publicKey;
        this.privateKey = privateKey;
    }

    public static AsymKeyPair copy(AsymKeyPair pair){
        return new AsymKeyPair(pair.getPublicKey(),pair.getPrivateKey());
    }

    public AsymKeyPair copy(){
        return copy(this);
    }


    public static void saveAsymKey(AsymKeyPair asymKey, OutputStream os) throws IOException {
        String pubKey = asymKey.getPublicKey();
        String priKey = asymKey.getPrivateKey();
        if (pubKey == null) {
            pubKey = "";
        }
        if (priKey == null) {
            priKey = "";
        }
        pubKey = Base64Obfuscator.encode(pubKey, true);
        priKey = Base64Obfuscator.encode(priKey, true);
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
        writer.write(pubKey);
        writer.newLine();
        writer.write(priKey);
        writer.newLine();
        writer.flush();
        os.flush();
    }

    public static AsymKeyPair loadAsymKey(InputStream is) throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(is, "UTF-8"));
        String pubKey = reader.readLine();
        String priKey = reader.readLine();
        pubKey = Base64Obfuscator.decode(pubKey);
        priKey = Base64Obfuscator.decode(priKey);
        if (pubKey == null || pubKey.isEmpty()) {
            pubKey = null;
        }
        if (priKey == null || priKey.isEmpty()) {
            priKey = null;
        }
        return new AsymKeyPair(pubKey, priKey);
    }
}
